maven/lib/dependabot/maven/version_parser.rb

```diff
@@ -0,0 +1,137 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "strscan"
+require "dependabot/maven/token_bucket"
+
+# See https://maven.apache.org/pom.html#Version_Order_Specification for details
+#
+module Dependabot
+  module Maven
+    class VersionParser
+      extend T::Sig
+      extend T::Helpers
+
+      ALPHA = -5
+      BETA = -4
+      MILESTONE = -3
+      RC = -2
+      SNAPSHOT = -1
+      SP = "sp"
+
+      sig { params(version: T.nilable(String)).returns(TokenBucket) }
+      def self.parse(version)
+        raise ArgumentError, "Malformed version string #{version}" if version.nil?
```

Comment:
this message will be "Malformed version string " since version `nil` coerces to `""`

Reply:
How about throwing the `MalformedVersionString` with the actual reason e.g. nil or empty string

Reply:
> this message will be "Malformed version string " since version `nil` coerces to `""`

Yes, I'm aware. I made it like that for consistency with what we currently have. The space helped me diagnose a bug when I was working on the bugs a few weeks ago

Reply:
I'll change it